### DISCUSSION TITLE
[@types/react] CSSProperties should honor csstype, fixing TS autocompletion

### DISCUSSION THREAD

**Discussion by marktoman** [DefinitelyTyped/DefinitelyTyped#66835] [Issues with a @types package] [answered]
Current definition:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/3e16768859508fd925a846cfaec7c8fce3b7702d/types/react/index.d.ts#L1669

This ruins Typescript autocompletion.

Csstype itself uses it everywhere with the TS fix:
https://github.com/frenic/csstype/blob/46694defae2cf3386218d0000490b0d0ac385aa6/index.d.ts#L11

Fix:
```diff
-    export interface CSSProperties extends CSS.Properties<string | number> {
+    export interface CSSProperties extends CSS.Properties<(string & {}) | number> {
```

[a comment by an automated account is not shown]

**Comment by marktoman** [comment]
Here's an example of broken autocompletion:

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/47689923/9ee7f8af-77de-40ad-be4a-053e742e2993)

It affects properties that accept string as well as union, like `verticalAlign`.

**Comment by eps1lon** [comment]
Makes sense. Would appreciate a PR. Let me know if anything is unclear.

**Comment by Semigradsky** [comment, marked as answer]
PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67583
